From 1a467684bd62fe585571b724ca8f93304e4db396 Mon Sep 17 00:00:00 2001
From: Roman Mazurak <roman.mazurak@infineon.com>
Date: Wed, 19 Feb 2025 23:43:35 +0200
Subject: [PATCH] Suppress build warnings

- Suppress pthread discovery for TF-M and warning
- Suppress GCC -Wformat-signedness flag discovery error message
   `Performing Test C_COMPILER_SUPPORT S_WFORMAT_SIGNEDNESS - Failed`

Change-Id: Ifb75192ffe1df6f72f5d6122f75133ca12c75f88
---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ba87c365..7ad2355d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,10 +120,6 @@ if(MBEDTLS_PYTHON_EXECUTABLE)
 
 endif()
 
-# We now potentially need to link all executables against PThreads, if available
-set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-set(THREADS_PREFER_PTHREAD_FLAG TRUE)
-find_package(Threads)
 
 # If this is the root project add longer list of available CMAKE_BUILD_TYPE values
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
@@ -210,10 +206,12 @@ if(CMAKE_COMPILER_IS_GNU)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wshadow")
     endif()
     if (GCC_VERSION VERSION_GREATER 5.0)
+        set(CMAKE_REQUIRED_QUIET ON)
         CHECK_C_COMPILER_FLAG("-Wformat-signedness" C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
         if(C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
         endif()
+        set(CMAKE_REQUIRED_QUIET OFF)
     endif()
     if (GCC_VERSION VERSION_GREATER 7.0 OR GCC_VERSION VERSION_EQUAL 7.0)
       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-overflow=2 -Wformat-truncation")
-- 
2.39.2.windows.1

